src/services/inplayer.account.service.ts

```diff
@@ -283,7 +284,8 @@ export const updatePersonalShelves: UpdatePersonalShelves = async (payload) => {
             await InPlayer.Account.updateWatchHistory(history.mediaid, history.progress);
           }
         });
-      } else {
+      }
+      if (!currentWatchHistoryIds?.includes(history.mediaid)) {
```

Comment:
Wait won't this prevent the progress from being updated?

Reply:
No, this checks if there is an `mediaid` that is not present in the store (currentWatchHistoryIds), then make a request with that id.

Reply:
I'm sorry, I still don't understand this... it looks like these 2 clauses are just doing the same thing now. Both ifs call `await InPlayer.Account.updateWatchHistory(history.mediaid, history.progress);` and they are essentially checking if externalData?.history contains an item with history.mediaid.

Reply:
The first one is only called when there is at least one item in the store and checks if the progress of some of the existing ones is updated, then call the endpoint. 
The newly added check, only calls the endpoint for new watch history items or if there is no watch history item at all (since the other one is only valid if there is at least one item).

Reply:
Didn't wanna spend time on refactoring this, just wanted to quickly fix the bug, but yeah, it will be much simpler this way. I will update it.